## Final Project Submission

Please fill out:
* Student name: Drake Fitzsimmons
* Student pace: Full-time
* Scheduled project review date/time: 3/23/21
* Instructor name: James Irving
* Blog post URL:


## Business Problem

## Obtain and Review Data

In [1]:
import os,glob
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df_basics = pd.read_csv('./zippedData/imdb.title.basics.csv.gz')
df_ratings = pd.read_csv('./zippedData/imdb.title.ratings.csv.gz')
df_gross = pd.read_csv('./zippedData/bom.movie_gross.csv.gz')

In [3]:
print('\n'+'---'*25)
print('df_basics preview')
display(df_basics)
print('\n'+'---'*25)
print('df_ratings preview')
display(df_ratings)
print('\n'+'---'*25)
print('df_gross preview')
display(df_gross)
print('\n')


---------------------------------------------------------------------------
df_basics preview


tconst                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                   NaN  
146143              NaN           Documentary  

[146144 rows x 6 columns]


---------------------------------------------------------------------------
df_ratings preview


tconst  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]


---------------------------------------------------------------------------
df_gross preview


title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

## Data Cleaning

### Combining Basics & Ratings
First, the dataframes must be combined in order to compare across different variables. We know that there is a common column between df_basics and df_ratings. Let's start there.

In [4]:
df_basics_ratings = df_basics.merge(df_ratings,how='inner',on='tconst')
df_basics_ratings

tconst                    primary_title              original_title  \
0      tt0063540                        Sunghursh                   Sunghursh   
1      tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2      tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3      tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4      tt0100275         The Wandering Soap Opera       La Telenovela Errante   
...          ...                              ...                         ...   
73851  tt9913084                 Diabolik sono io            Diabolik sono io   
73852  tt9914286                Sokagin Çocuklari           Sokagin Çocuklari   
73853  tt9914642                        Albatross                   Albatross   
73854  tt9914942       La vida sense la Sara Amat  La vida sense la Sara Amat   
73855  tt9916160                       Drømmeland                  Drømmeland   

       start_year  runtime_minutes                genres  averagerating  \
0            2013            175.0    Action,Crime,Drama            7.0   
1            2019            114.0       Biography,Drama            7.2   
2            2018            122.0                 Drama            6.9   
3            2018              NaN          Comedy,Drama            6.1   
4            2017             80.0  Comedy,Drama,Fantasy            6.5   
...           ...              ...                   ...            ...   
73851        2019             75.0           Documentary            6.2   
73852        2019             98.0          Drama,Family            8.7   
73853        2017              NaN           Documentary            8.5   
73854        2019              NaN                   NaN            6.6   
73855        2019             72.0           Documentary            6.5   

       numvotes  
0            77  
1            43  
2          4517  
3            13  
4           119  
...         ...  
73851         6  
73852       136  
73853         8  
73854         5  
73855        11  

[73856 rows x 8 columns]

df_basics_ratings is a new dataframe that combined df_basics and df_ratings. The total rows in df_basics_ratings is equal to the total number of rows in df_ratings. Presumably every row in df_ratings had a corresponding row in df_basics and all rows unique to df_basics have been removed. Let's check.

In [5]:
print(df_basics_ratings.isna().sum())
df_dup_check = df_basics_ratings[df_basics_ratings.duplicated('tconst')]
df_dup_check

tconst                0
primary_title         0
original_title        0
start_year            0
runtime_minutes    7620
genres              804
averagerating         0
numvotes              0
dtype: int64


Empty DataFrame
Columns: [tconst, primary_title, original_title, start_year, runtime_minutes, genres, averagerating, numvotes]
Index: []

In [6]:
df_dup_check = df_basics[df_basics.duplicated('tconst')]
df_dup_check

Empty DataFrame
Columns: [tconst, primary_title, original_title, start_year, runtime_minutes, genres]
Index: []

In [7]:
df_dup_check = df_ratings[df_ratings.duplicated('tconst')]
df_dup_check

Empty DataFrame
Columns: [tconst, averagerating, numvotes]
Index: []

We can see that there are no null values in the averagerating and numvotes columns. Also there are no duplicate tconst values in df_basics_ratings, df_basics, or df_ratings. This confirms that we removed rows from df_basics that did not have a corresponding tconst in df_ratings.

### Combining Revenue
Next we must combine df_gross to our newly created df_basics_ratings. Let's find a common column to combine with.

In [8]:
print('\n'+'---'*25)
print('df_basics_ratings preview')
display(df_basics)
print('\n'+'---'*25)
print('df_gross preview')
display(df_gross)


---------------------------------------------------------------------------
df_basics_ratings preview


tconst                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                   NaN  
146143              NaN           Documentary  

[146144 rows x 6 columns]


---------------------------------------------------------------------------
df_gross preview


title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

Common columns include title and year. Unfortunately in df_basics_ratings, there are two versions of title: primary_title and original_title. Ultimately we will have to check to see if titles in df_gross match with primary_title or original_title. But first let's see if there are any duplicates in df_gross that we should deal with.

In [9]:
df_dup_check = None
df_dup_check = df_gross[df_gross.duplicated('title', keep = False)]
df_dup_check

title  studio  domestic_gross foreign_gross  year
317   Bluebeard  Strand         33500.0          5200  2010
3045  Bluebeard   WGUSA         43100.0           NaN  2017

There are two movies with the title Bluebeard mad by different studios in different years. There is no need to worry though, because we can merge df_gross with df_basics_ratings using two criteria, title and year. Since the years are different there shouldn't be a problem.

Let's merge df_gross with df_basics_ratings.

In [10]:
df_combined = df_basics_ratings.merge(df_gross,how='inner',
                                      left_on = ['primary_title','start_year'],
                                      right_on = ['title','year'])
df_combined

tconst                    primary_title  \
0     tt0315642                            Wazir   
1     tt0337692                      On the Road   
2     tt0359950  The Secret Life of Walter Mitty   
3     tt0365907      A Walk Among the Tombstones   
4     tt0369610                   Jurassic World   
...         ...                              ...   
1842  tt8290698               The Spy Gone North   
1843  tt8404272           How Long Will I Love U   
1844  tt8427036                  Helicopter Eela   
1845  tt9078374                      Last Letter   
1846  tt9151704        Burn the Stage: The Movie   

                       original_title  start_year  runtime_minutes  \
0                               Wazir        2016            103.0   
1                         On the Road        2012            124.0   
2     The Secret Life of Walter Mitty        2013            114.0   
3         A Walk Among the Tombstones        2014            114.0   
4                      Jurassic World        2015            124.0   
...                               ...         ...              ...   
1842                          Gongjak        2018            137.0   
1843            Chao shi kong tong ju        2018            101.0   
1844                  Helicopter Eela        2018            135.0   
1845                   Ni hao, Zhihua        2018            114.0   
1846        Burn the Stage: The Movie        2018             84.0   

                       genres  averagerating  numvotes  \
0          Action,Crime,Drama            7.1     15378   
1     Adventure,Drama,Romance            6.1     37886   
2      Adventure,Comedy,Drama            7.3    275300   
3          Action,Crime,Drama            6.5    105116   
4     Action,Adventure,Sci-Fi            7.0    539338   
...                       ...            ...       ...   
1842                    Drama            7.2      1620   
1843                  Romance            6.5       607   
1844                    Drama            5.4       673   
1845            Drama,Romance            6.4       322   
1846        Documentary,Music            8.8      2067   

                                title     studio  domestic_gross  \
0                               Wazir    Relbig.       1100000.0   
1                         On the Road        IFC        744000.0   
2     The Secret Life of Walter Mitty        Fox      58200000.0   
3         A Walk Among the Tombstones       Uni.      26300000.0   
4                      Jurassic World       Uni.     652300000.0   
...                               ...        ...             ...   
1842               The Spy Gone North         CJ        501000.0   
1843           How Long Will I Love U      WGUSA        747000.0   
1844                  Helicopter Eela       Eros         72000.0   
1845                      Last Letter         CL        181000.0   
1846        Burn the Stage: The Movie  Trafalgar       4200000.0   

     foreign_gross  year  
0              NaN  2016  
1          8000000  2012  
2        129900000  2013  
3         26900000  2014  
4          1,019.4  2015  
...            ...   ...  
1842           NaN  2018  
1843      82100000  2018  
1844           NaN  2018  
1845           NaN  2018  
1846      16100000  2018  

[1847 rows x 13 columns]

Great, now we have a combined dataframe. But we only combined based on primary_title. Let's ensure that there weren't any movies which the original title would match with a movie in df_gross.

In [11]:
df_unequal_titles = df_basics_ratings.loc[df_basics_ratings['primary_title'] != df_basics_ratings['original_title']]
df_unequal_titles

tconst                      primary_title  \
1      tt0066787    One Day Before the Rainy Season   
4      tt0100275           The Wandering Soap Opera   
8      tt0154039               So Much for Justice!   
10     tt0162942       Children of the Green Dragon   
12     tt0176694                 The Tragedy of Man   
...          ...                                ...   
73804  tt9875852                      The House Elf   
73824  tt9894394  Upin & Ipin: The Lone Gibbon Kris   
73829  tt9899840                        Auntie Frog   
73830  tt9899850                      The Agitation   
73831  tt9899860     Watching This Movie Is a Crime   

                           original_title  start_year  runtime_minutes  \
1                         Ashad Ka Ek Din        2019            114.0   
4                   La Telenovela Errante        2017             80.0   
8                          Oda az igazság        2010            100.0   
10               A zöld sárkány gyermekei        2010             89.0   
12                    Az ember tragédiája        2011            160.0   
...                                   ...         ...              ...   
73804                             Domovoy        2019             90.0   
73824  Upin & Ipin: Keris Siamang Tunggal        2019            100.0   
73829                   Khaleh Ghurbagheh        2018             81.0   
73830                           Ashoftegi        2019              NaN   
73831              Didan in film jorm ast        2019            100.0   

                        genres  averagerating  numvotes  
1              Biography,Drama            7.2        43  
4         Comedy,Drama,Fantasy            6.5       119  
8                      History            4.6        64  
10                       Drama            6.9       120  
12     Animation,Drama,History            7.8       584  
...                        ...            ...       ...  
73804    Comedy,Family,Fantasy            5.8        50  
73824                Animation            8.1       301  
73829  Adventure,Comedy,Family            6.2         6  
73830           Drama,Thriller            4.9        14  
73831           Drama,Thriller            8.1         7  

[12173 rows x 8 columns]

There are 12,173 movies where the primary_title does not equal the original_title. Let's see which of these movies match with df_gross.

In [12]:
df_unequal_titles = df_unequal_titles.merge(df_gross,how='inner',
                                      left_on = ['original_title','start_year'],
                                      right_on = ['title','year'])
df_unequal_titles

tconst                    primary_title  \
0   tt1139592             Jeepers Creepers III   
1   tt1433813                           Hubble   
2   tt1477076        Saw 3D: The Final Chapter   
3   tt1620719                         Fearless   
4   tt1629376              Seven Sins Forgiven   
5   tt3148502                    The Spectacle   
6   tt3173910        She Smiles, She's Snared!   
7   tt3381068                      Gnome Alone   
8   tt3595298     Found a Treasure Called Love   
9   tt3698118                    Gangnam Blues   
10  tt3833746    The Visit: An Alien Encounter   
11  tt4410000              Luis and the Aliens   
12  tt4500922                   The Death Cure   
13  tt5162658                             Gold   
14  tt6217804         Boo 2! A Madea Halloween   
15  tt6588966                           Hiccup   
16  tt7690670                         SuperFly   
17  tt8549902  Oolong Courtyard: KungFu School   

                            original_title  start_year  runtime_minutes  \
0                       Jeepers Creepers 3        2017            100.0   
1                                Hubble 3D        2010             44.0   
2                                   Saw 3D        2010             90.0   
3                                  Dabangg        2010            126.0   
4                             7 Khoon Maaf        2011            137.0   
5                                  Tamasha        2015            139.0   
6                         Hasee Toh Phasee        2014            141.0   
7                                   Legend        2015             94.0   
8                     Prem Ratan Dhan Payo        2015            164.0   
9                             Gangnam 1970        2015            135.0   
10                               The Visit        2015             83.0   
11                       Luis & the Aliens        2018             86.0   
12             Maze Runner: The Death Cure        2018            143.0   
13                                     Oro        2017            103.0   
14  Tyler Perry's Boo 2! A Madea Halloween        2017            101.0   
15                                  Hichki        2018            116.0   
16                                Superfly        2018            116.0   
17                        Oolong Courtyard        2018            103.0   

                        genres  averagerating  numvotes  \
0        Action,Horror,Mystery            4.1     14586   
1                  Documentary            7.7      3945   
2         Crime,Horror,Mystery            5.6     83532   
3          Action,Comedy,Crime            6.3     25201   
4       Drama,Mystery,Thriller            6.1      4948   
5         Comedy,Drama,Romance            7.2     19901   
6               Comedy,Romance            6.8     12406   
7                       Horror            3.5       303   
8          Action,Drama,Family            4.6     15753   
9                 Action,Drama            6.3      1291   
10                 Documentary            6.0       661   
11  Adventure,Animation,Comedy            6.0      1355   
12      Action,Sci-Fi,Thriller            6.2     91286   
13     Adventure,Drama,History            5.6      1103   
14         Comedy,Drama,Horror            3.8      3383   
15                Comedy,Drama            7.5      7418   
16       Action,Crime,Thriller            5.0      4753   
17                      Comedy            4.6        61   

                                     title   studio  domestic_gross  \
0                       Jeepers Creepers 3    Scre.       2300000.0   
1                                Hubble 3D       WB      52400000.0   
2                                   Saw 3D      LGF      45700000.0   
3                                  Dabangg     Eros       1300000.0   
4                             7 Khoon Maaf      UTV        270000.0   
5                                  Tamasha      UTV       2100000.0   
6          

There are 18 movies that match based on original_title and not primary_title. We should include these in the combined dataframe.

In [13]:
dataframes = [df_combined, df_unequal_titles]
df_combined = pd.concat(dataframes)
df_combined

tconst                    primary_title  \
0   tt0315642                            Wazir   
1   tt0337692                      On the Road   
2   tt0359950  The Secret Life of Walter Mitty   
3   tt0365907      A Walk Among the Tombstones   
4   tt0369610                   Jurassic World   
..        ...                              ...   
13  tt5162658                             Gold   
14  tt6217804         Boo 2! A Madea Halloween   
15  tt6588966                           Hiccup   
16  tt7690670                         SuperFly   
17  tt8549902  Oolong Courtyard: KungFu School   

                            original_title  start_year  runtime_minutes  \
0                                    Wazir        2016            103.0   
1                              On the Road        2012            124.0   
2          The Secret Life of Walter Mitty        2013            114.0   
3              A Walk Among the Tombstones        2014            114.0   
4                           Jurassic World        2015            124.0   
..                                     ...         ...              ...   
13                                     Oro        2017            103.0   
14  Tyler Perry's Boo 2! A Madea Halloween        2017            101.0   
15                                  Hichki        2018            116.0   
16                                Superfly        2018            116.0   
17                        Oolong Courtyard        2018            103.0   

                     genres  averagerating  numvotes  \
0        Action,Crime,Drama            7.1     15378   
1   Adventure,Drama,Romance            6.1     37886   
2    Adventure,Comedy,Drama            7.3    275300   
3        Action,Crime,Drama            6.5    105116   
4   Action,Adventure,Sci-Fi            7.0    539338   
..                      ...            ...       ...   
13  Adventure,Drama,History            5.6      1103   
14      Comedy,Drama,Horror            3.8      3383   
15             Comedy,Drama            7.5      7418   
16    Action,Crime,Thriller            5.0      4753   
17                   Comedy            4.6        61   

                                     title   studio  domestic_gross  \
0                                    Wazir  Relbig.       1100000.0   
1                              On the Road      IFC        744000.0   
2          The Secret Life of Walter Mitty      Fox      58200000.0   
3              A Walk Among the Tombstones     Uni.      26300000.0   
4                           Jurassic World     Uni.     652300000.0   
..                                     ...      ...             ...   
13                                     Oro     Sony          5500.0   
14  Tyler Perry's Boo 2! A Madea Halloween      LGF      47300000.0   
15                                  Hichki     Yash        330000.0   
16                                Superfly     Sony      20500000.0   
17                        Oolong Courtyard       CL         37700.0   

   foreign_gross  year  
0            NaN  2016  
1        8000000  2012  
2      129900000  2013  
3       26900000  2014  
4        1,019.4  2015  
..           ...   ...  
13           NaN  2017  
14       1000000  2017  
15       3900000  2018  
16        236000  2018  
17           NaN  2018  

[1865 rows x 13 columns]

Now we have all the data we need combined into 1 dataframe. Let's clean up our column names and then dig into any data we need to clean up.

### Removing Unnecessary Columns
There are columns which are now duplicates and columns that don't really mean much to us. Let's remove tconst because it doesn't have relevant meaning and title and year because they are just duplicate columns now.

In [14]:
del df_combined['tconst']
del df_combined['title']
del df_combined['year']
df_combined

primary_title                          original_title  \
0                             Wazir                                   Wazir   
1                       On the Road                             On the Road   
2   The Secret Life of Walter Mitty         The Secret Life of Walter Mitty   
3       A Walk Among the Tombstones             A Walk Among the Tombstones   
4                    Jurassic World                          Jurassic World   
..                              ...                                     ...   
13                             Gold                                     Oro   
14         Boo 2! A Madea Halloween  Tyler Perry's Boo 2! A Madea Halloween   
15                           Hiccup                                  Hichki   
16                         SuperFly                                Superfly   
17  Oolong Courtyard: KungFu School                        Oolong Courtyard   

    start_year  runtime_minutes                   genres  averagerating  \
0         2016            103.0       Action,Crime,Drama            7.1   
1         2012            124.0  Adventure,Drama,Romance            6.1   
2         2013            114.0   Adventure,Comedy,Drama            7.3   
3         2014            114.0       Action,Crime,Drama            6.5   
4         2015            124.0  Action,Adventure,Sci-Fi            7.0   
..         ...              ...                      ...            ...   
13        2017            103.0  Adventure,Drama,History            5.6   
14        2017            101.0      Comedy,Drama,Horror            3.8   
15        2018            116.0             Comedy,Drama            7.5   
16        2018            116.0    Action,Crime,Thriller            5.0   
17        2018            103.0                   Comedy            4.6   

    numvotes   studio  domestic_gross foreign_gross  
0      15378  Relbig.       1100000.0           NaN  
1      37886      IFC        744000.0       8000000  
2     275300      Fox      58200000.0     129900000  
3     105116     Uni.      26300000.0      26900000  
4     539338     Uni.     652300000.0       1,019.4  
..       ...      ...             ...           ...  
13      1103     Sony          5500.0           NaN  
14      3383      LGF      47300000.0       1000000  
15      7418     Yash        330000.0       3900000  
16      4753     Sony      20500000.0        236000  
17        61       CL         37700.0           NaN  

[1865 rows x 10 columns]

Great, that's easier to work with. Now let's take a deeper look into what data we actually have.

### Removing Duplicates
Now that we have all our data in 1 place let's see if we have any duplicate rows.

In [26]:
df_dup_check = None
df_dup_check = df_combined[df_combined.duplicated('primary_title', keep = False)]
df_dup_check

primary_title              original_title  start_year  \
140   The Bounty Hunter           The Bounty Hunter        2010   
141   The Bounty Hunter           The Bounty Hunter        2010   
167            Big Eyes                    Big Eyes        2014   
168            Big Eyes                    Big Eyes        2014   
169           Burlesque                   Burlesque        2010   
170           Burlesque                   Burlesque        2010   
195              Jai Ho                      Jai Ho        2014   
196              Jai Ho                      Jai Ho        2014   
225           Leap Year                   Leap Year        2010   
226           Leap Year                Año bisiesto        2010   
272         The Tempest                 The Tempest        2010   
273         The Tempest                 The Tempest        2010   
319               Cyrus                       Cyrus        2010   
320               Cyrus                       Cyrus        2010   
335      I'm Still Here              I'm Still Here        2010   
336      I'm Still Here              I'm Still Here        2010   
497       A Better Life               A Better Life        2011   
498       A Better Life           Une vie meilleure        2011   
547           Abduction                   Abduction        2011   
548           Abduction                   Abduction        2011   
616          The Artist                  The Artist        2011   
617          The Artist                  The Artist        2011   
743                Gone                        Gone        2012   
744                Gone                        Gone        2012   
774        The Darkness                The Darkness        2016   
775        The Darkness               Las tinieblas        2016   
781           Spotlight                   Spotlight        2015   
782           Spotlight                   Spotlight        2015   
786            The Call                    The Call        2013   
787            The Call                     Lokroep        2013   
932            The Walk                    The Walk        2015   
933            The Walk                    The Walk        2015   
993          True Story                  True Story        2015   
994          True Story                  True Story        2015   
1024          Homefront                   Homefront        2013   
1025          Homefront  The Things We Leave Behind        2013   
1069               Coco                        Coco        2017   
1070               Coco                        Coco        2017   
1294         Extraction                  Extraction        2015   
1295         Extraction                  Extraction        2015   
1306         The Forest                  The Forest        2016   
1307         The Forest                  The Forest        2016   
1421           Stronger                    Stronger        2017   
1422           Stronger                    Stronger        2017   
1526             Denial                      Denial        2016   
1527             Denial                      Denial        2016   
1611             Dancer                      Dancer        2016   
1612             Dancer                      Dancer        2016   
1615               Gold                         Oro        2017   
1787    The Negotiation             The Negotiation        2018   
1788    The Negotiation             The Negotiation        2018   
13                 Gold                         Oro        2017   

      runtime_minutes                       genres  averagerating  numvotes  \
140             110.0        Action,Comedy,Romance            5.6    112444   
141               NaN                          NaN            6.3        29   
167             106.0        Biography,Crime,Drama            7.0     77447   
168               NaN                  Documentary            7.2        43   
169             119.0          Drama,Music,Musical            6

As you can see, there are quiet a few movies with the same title. But it's possible that some movies would have the same title. It's unlikely that the a studio would produce 2 movies with the same title in the same year. Let's make a new column to include title, year, and studio. 

In [35]:
df_combined['title_year_studio'] = df_combined['primary_title'].astype(str)\
                                    + df_combined['start_year'].astype(str)\
                                    + df_combined['studio'].astype(str)
df_combined

primary_title                          original_title  \
0                             Wazir                                   Wazir   
1                       On the Road                             On the Road   
2   The Secret Life of Walter Mitty         The Secret Life of Walter Mitty   
3       A Walk Among the Tombstones             A Walk Among the Tombstones   
4                    Jurassic World                          Jurassic World   
..                              ...                                     ...   
13                             Gold                                     Oro   
14         Boo 2! A Madea Halloween  Tyler Perry's Boo 2! A Madea Halloween   
15                           Hiccup                                  Hichki   
16                         SuperFly                                Superfly   
17  Oolong Courtyard: KungFu School                        Oolong Courtyard   

    start_year  runtime_minutes                   genres  averagerating  \
0         2016            103.0       Action,Crime,Drama            7.1   
1         2012            124.0  Adventure,Drama,Romance            6.1   
2         2013            114.0   Adventure,Comedy,Drama            7.3   
3         2014            114.0       Action,Crime,Drama            6.5   
4         2015            124.0  Action,Adventure,Sci-Fi            7.0   
..         ...              ...                      ...            ...   
13        2017            103.0  Adventure,Drama,History            5.6   
14        2017            101.0      Comedy,Drama,Horror            3.8   
15        2018            116.0             Comedy,Drama            7.5   
16        2018            116.0    Action,Crime,Thriller            5.0   
17        2018            103.0                   Comedy            4.6   

    numvotes   studio  domestic_gross foreign_gross  \
0      15378  Relbig.       1100000.0           NaN   
1      37886      IFC        744000.0       8000000   
2     275300      Fox      58200000.0     129900000   
3     105116     Uni.      26300000.0      26900000   
4     539338     Uni.     652300000.0       1,019.4   
..       ...      ...             ...           ...   
13      1103     Sony          5500.0           NaN   
14      3383      LGF      47300000.0       1000000   
15      7418     Yash        330000.0       3900000   
16      4753     Sony      20500000.0        236000   
17        61       CL         37700.0           NaN   

                         title_year_studio  
0                         Wazir2016Relbig.  
1                       On the Road2012IFC  
2   The Secret Life of Walter Mitty2013Fox  
3      A Walk Among the Tombstones2014Uni.  
4                   Jurassic World2015Uni.  
..                                     ...  
13                            Gold2017Sony  
14         Boo 2! A Madea Halloween2017LGF  
15                          Hiccup2018Yash  
16                        SuperFly2018Sony  
17   Oolong Courtyard: KungFu School2018CL  

[1865 rows x 11 columns]

In [39]:
df_dup_check = None
df_dup_check = df_combined[df_combined.duplicated('title_year_studio', keep = False)]
df_dup_check

primary_title              original_title  start_year  \
140   The Bounty Hunter           The Bounty Hunter        2010   
141   The Bounty Hunter           The Bounty Hunter        2010   
167            Big Eyes                    Big Eyes        2014   
168            Big Eyes                    Big Eyes        2014   
169           Burlesque                   Burlesque        2010   
170           Burlesque                   Burlesque        2010   
195              Jai Ho                      Jai Ho        2014   
196              Jai Ho                      Jai Ho        2014   
225           Leap Year                   Leap Year        2010   
226           Leap Year                Año bisiesto        2010   
272         The Tempest                 The Tempest        2010   
273         The Tempest                 The Tempest        2010   
319               Cyrus                       Cyrus        2010   
320               Cyrus                       Cyrus        2010   
335      I'm Still Here              I'm Still Here        2010   
336      I'm Still Here              I'm Still Here        2010   
497       A Better Life               A Better Life        2011   
498       A Better Life           Une vie meilleure        2011   
547           Abduction                   Abduction        2011   
548           Abduction                   Abduction        2011   
616          The Artist                  The Artist        2011   
617          The Artist                  The Artist        2011   
743                Gone                        Gone        2012   
744                Gone                        Gone        2012   
774        The Darkness                The Darkness        2016   
775        The Darkness               Las tinieblas        2016   
781           Spotlight                   Spotlight        2015   
782           Spotlight                   Spotlight        2015   
786            The Call                    The Call        2013   
787            The Call                     Lokroep        2013   
932            The Walk                    The Walk        2015   
933            The Walk                    The Walk        2015   
993          True Story                  True Story        2015   
994          True Story                  True Story        2015   
1024          Homefront                   Homefront        2013   
1025          Homefront  The Things We Leave Behind        2013   
1069               Coco                        Coco        2017   
1070               Coco                        Coco        2017   
1294         Extraction                  Extraction        2015   
1295         Extraction                  Extraction        2015   
1306         The Forest                  The Forest        2016   
1307         The Forest                  The Forest        2016   
1421           Stronger                    Stronger        2017   
1422           Stronger                    Stronger        2017   
1526             Denial                      Denial        2016   
1527             Denial                      Denial        2016   
1611             Dancer                      Dancer        2016   
1612             Dancer                      Dancer        2016   
1787    The Negotiation             The Negotiation        2018   
1788    The Negotiation             The Negotiation        2018   

      runtime_minutes                       genres  averagerating  numvotes  \
140             110.0        Action,Comedy,Romance            5.6    112444   
141               NaN                          NaN            6.3        29   
167             106.0        Biography,Crime,Drama            7.0     77447   
168               NaN                  Documentary            7.2        43   
169             119.0          Drama,Music,Musical            6.4     71021   
170               NaN                        Drama            7.0        45   
195             135.0                 A

In [38]:
df_combined.sort_values(by='numvotes')
df_combined.sort_values(by='primary_title')
df_dup_check = None
df_dup_check = df_combined[df_combined.duplicated('primary_title', keep = False)]
df_dup_check.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52 entries, 140 to 13
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   primary_title      52 non-null     object 
 1   original_title     52 non-null     object 
 2   start_year         52 non-null     int64  
 3   runtime_minutes    48 non-null     float64
 4   genres             50 non-null     object 
 5   averagerating      52 non-null     float64
 6   numvotes           52 non-null     int64  
 7   studio             52 non-null     object 
 8   domestic_gross     52 non-null     float64
 9   foreign_gross      35 non-null     object 
 10  title_year_studio  52 non-null     object 
dtypes: float64(3), int64(2), object(6)
memory usage: 4.9+ KB


In [23]:
df_combined.info()
df_combined.isna().sum()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1865 entries, 0 to 17
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   primary_title    1865 non-null   object 
 1   original_title   1865 non-null   object 
 2   start_year       1865 non-null   int64  
 3   runtime_minutes  1861 non-null   float64
 4   genres           1863 non-null   object 
 5   averagerating    1865 non-null   float64
 6   numvotes         1865 non-null   int64  
 7   studio           1863 non-null   object 
 8   domestic_gross   1855 non-null   float64
 9   foreign_gross    1280 non-null   object 
dtypes: float64(3), int64(2), object(5)
memory usage: 160.3+ KB


primary_title        0
original_title       0
start_year           0
runtime_minutes      4
genres               2
averagerating        0
numvotes             0
studio               2
domestic_gross      10
foreign_gross      585
dtype: int64

## Extra Stuff

In [19]:
df_basics.info()
df_basics.isna().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


tconst                 0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In [21]:
df_combined.describe()

start_year  runtime_minutes  averagerating      numvotes  \
count  1865.000000      1861.000000    1865.000000  1.865000e+03   
mean   2013.982842       110.785062       6.415979  9.094046e+04   
std       2.519607        19.945410       1.008649  1.496859e+05   
min    2010.000000        25.000000       1.600000  6.000000e+00   
25%    2012.000000        96.000000       5.800000  7.384000e+03   
50%    2014.000000       107.000000       6.500000  3.430600e+04   
75%    2016.000000       122.000000       7.100000  1.049990e+05   
max    2018.000000       189.000000       8.800000  1.841066e+06   

       domestic_gross  
count    1.855000e+03  
mean     4.253601e+07  
std      7.694477e+07  
min      3.000000e+02  
25%      5.775000e+05  
50%      1.070000e+07  
75%      5.190000e+07  
max      7.001000e+08

## xxx

In [ ]:
df_dup_check = None
df_dup_check = df_basics[df_basics.duplicated('primary_title', keep = False)]
df_dup_check

In [ ]:
df_dup_basics = df_basics_ratings[df_basics_ratings.duplicated('original_title', keep = False)]
df_dup_basics

In [ ]:
df_gross.describe()

In [ ]:
df_combined = df_basics_ratings.merge(df_gross,how='inner',
                                      left_on = ['primary_title','start_year'],
                                      right_on = ['title','year'])
df_combined

In [ ]:
df_unequal_titles = df_combined.loc[df_combined['primary_title'] != df_combined['original_title']]
df_unequal_titles

In [ ]:
df_unequal_titles[df_unequal_titles['primary_title'] == 'Gold']

In [21]:
df_basics_ratings = df_basics.merge(df_ratings,how='inner',on='tconst')
df_basics_ratings

tconst                    primary_title              original_title  \
0      tt0063540                        Sunghursh                   Sunghursh   
1      tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2      tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3      tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4      tt0100275         The Wandering Soap Opera       La Telenovela Errante   
...          ...                              ...                         ...   
73851  tt9913084                 Diabolik sono io            Diabolik sono io   
73852  tt9914286                Sokagin Çocuklari           Sokagin Çocuklari   
73853  tt9914642                        Albatross                   Albatross   
73854  tt9914942       La vida sense la Sara Amat  La vida sense la Sara Amat   
73855  tt9916160                       Drømmeland                  Drømmeland   

       start_year  runtime_minutes                genres  averagerating  \
0            2013            175.0    Action,Crime,Drama            7.0   
1            2019            114.0       Biography,Drama            7.2   
2            2018            122.0                 Drama            6.9   
3            2018              NaN          Comedy,Drama            6.1   
4            2017             80.0  Comedy,Drama,Fantasy            6.5   
...           ...              ...                   ...            ...   
73851        2019             75.0           Documentary            6.2   
73852        2019             98.0          Drama,Family            8.7   
73853        2017              NaN           Documentary            8.5   
73854        2019              NaN                   NaN            6.6   
73855        2019             72.0           Documentary            6.5   

       numvotes  
0            77  
1            43  
2          4517  
3            13  
4           119  
...         ...  
73851         6  
73852       136  
73853         8  
73854         5  
73855        11  

[73856 rows x 8 columns]

## Visualizations